azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -74,44 +74,58 @@ def __init__(self, wire_client, full_goal_state=False, base_incarnation=None):
+            reason = 'new incarnation'
+
+        if not fetch_full_goal_state:
             self.hosting_env = None
             self.shared_conf = None
             self.certs = None
             self.ext_conf = None
             self.remote_access = None
             return
 
-        logger.info('Fetching new goal state [incarnation {0}]', self.incarnation)
+        logger.info('Fetching new goal state [incarnation {0} ({1})]', self.incarnation, reason)
 
-        uri = findtext(xml_doc, "HostingEnvironmentConfig")
-        xml_text = wire_client.fetch_config(uri, wire_client.get_header())
-        self.hosting_env = HostingEnv(xml_text)
-
-        uri = findtext(xml_doc, "SharedConfig")
-        xml_text = wire_client.fetch_config(uri, wire_client.get_header())
-        self.shared_conf = SharedConfig(xml_text)
+        try:
+            uri = findtext(xml_doc, "HostingEnvironmentConfig")
+            xml_text = wire_client.fetch_config(uri, wire_client.get_header())
+            self.hosting_env = HostingEnv(xml_text)
 
-        uri = findtext(xml_doc, "Certificates")
-        if uri is None:
-            self.certs = None
-        else:
-            xml_text = wire_client.fetch_config(uri, wire_client.get_header_for_cert())
-            self.certs = Certificates(xml_text)
-
-        uri = findtext(xml_doc, "ExtensionsConfig")
-        if uri is None:
-            self.ext_conf = ExtensionsConfig(None)
-        else:
+            uri = findtext(xml_doc, "SharedConfig")
             xml_text = wire_client.fetch_config(uri, wire_client.get_header())
-            self.ext_conf = ExtensionsConfig(xml_text)
+            self.shared_conf = SharedConfig(xml_text)
 
-        uri = findtext(container, "RemoteAccessInfo")
-        if uri is None:
-            self.remote_access = None
-        else:
-            xml_text = wire_client.fetch_config(uri, wire_client.get_header_for_cert())
-            self.remote_access = RemoteAccess(xml_text)
+            uri = findtext(xml_doc, "Certificates")
+            if uri is None:
+                self.certs = None
+            else:
+                xml_text = wire_client.fetch_config(uri, wire_client.get_header_for_cert())
+                self.certs = Certificates(xml_text)
+
+            uri = findtext(xml_doc, "ExtensionsConfig")
+            if uri is None:
+                self.ext_conf = ExtensionsConfig(None)
+            else:
+                xml_text = wire_client.fetch_config(uri, wire_client.get_header())
+                self.ext_conf = ExtensionsConfig(xml_text)
+
+            uri = findtext(container, "RemoteAccessInfo")
+            if uri is None:
+                self.remote_access = None
+            else:
+                xml_text = wire_client.fetch_config(uri, wire_client.get_header_for_cert())
+                self.remote_access = RemoteAccess(xml_text)
+        except Exception as e:
+            logger.warn("Fetching the goal state failed: {0}", ustr(e))
```

Comment:
Just a minor comment - Would the more accurate message be "Fetching the config files failed"?

Reply:
discussed offline. The current exception sounds correct.